server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/daemon/CheckRunningJob.java

```diff
@@ -54,51 +62,86 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
+        handleGeneralHeartTimeoutJobs(size, heartTimeoutPeriod);
+        if (taskFrameworkProperties.getRunMode().isProcess()) {
+            handleNotLocalAndProcessIsNotExitsJobs(size, heartTimeoutPeriod);
+        }
+    }
+
+    private void handleGeneralHeartTimeoutJobs(int size, int heartTimeoutPeriod) {
         // find heart timeout job
         Page<JobEntity> jobs = getConfiguration().getTaskFrameworkService()
                 .findHeartTimeTimeoutJobs(heartTimeoutPeriod, 0, size);
-        jobs.forEach(this::handleJobRetryingOrCanceled);
-
-    }
-
-    private void handleJobRetryingOrCanceled(JobEntity a) {
-        getConfiguration().getTransactionManager().doInTransactionWithoutResult(() -> {
-            doHandleJobRetryingOrFailed(a);
-        });
-
+        for (JobEntity j : jobs) {
+            SilentExecutor.executeSafely("handleJobRetryingOrFailed", () -> {
+                handleJobRetryingOrFailed(j, a -> {
+                    try {
+                        // destroy executor
+                        getConfiguration().getJobDispatcher().destroy(JobIdentity.of(a.getId()));
+                    } catch (JobException e) {
+                        throw new TaskRuntimeException(e);
+                    }
+                });
+            });
+        }
     }
 
-    private void doHandleJobRetryingOrFailed(JobEntity a) {
-        // destroy executor
-        try {
-            getConfiguration().getJobDispatcher().destroy(JobIdentity.of(a.getId()));
-        } catch (JobException e) {
-            throw new TaskRuntimeException(e);
+    private void handleNotLocalAndProcessIsNotExitsJobs(int size, int heartTimeoutPeriod) {
+        // ip has been changed docker restart and process has been interrupted,
+        // so we should find jobs which heart timeout and not running in local
+        Page<JobEntity> jobs = getConfiguration().getTaskFrameworkService()
+                .findHeartTimeTimeoutNotLocalJobs(heartTimeoutPeriod, 0, size);
+
+        List<JobEntity> jobList = jobs.stream().filter(a -> {
+            ExecutorIdentifier ei = ExecutorIdentifierParser.parser(a.getExecutorIdentifier());
+            return !(HttpUtil.isConnectable(ei.getHost(), ei.getPort(),
+                    getConfiguration().getTaskFrameworkProperties().getCheckOdcServerCanBeConnectedTimes()));
+        }).collect(Collectors.toList());
+
+        for (JobEntity j : jobList) {
+            SilentExecutor.executeSafely("handleJobRetryingOrFailed", () -> {
+                handleJobRetryingOrFailed(j, a -> {
+                    int rows = getConfiguration().getTaskFrameworkService().updateExecutorToDestroyed(a.getId());
+                    if (rows > 0) {
+                        log.info("Executor is not exists, update job executor to destroyed, jobId={}", a.getId());
+                    } else {
+                        throw new TaskRuntimeException("update executor to destroyed failed, jobId=" + a.getId());
```

Comment:
who will catch this exception?

Reply:
exception will be catch in `getConfiguration().getTransactionManager().doInTransactionWithoutResult` and transaction will be rollback